Gemfile

```diff
@@ -13,3 +13,6 @@ gem 'jruby-openssl', :platforms => :jruby
 gem 'mini_mime'
 
 gem 'byebug', :platforms => :mri
+
+gem "io-wait", ">= 0.2.2.pre1"
+gem "strscan", ">= 3.0.2.pre1"
```

Comment:
Ah JRuby needs it, and `strscan` is pulled in as a gem from one of the net-* gems I guess:
https://github.com/mikel/mail/runs/6114147775?check_suite_focus=true

`io-wait` shouldn't be a dependency of any of the `net-*` gems anymore, but unsure if there were releases with that.

Reply:
Yeah, JRuby needs the prereleases but Bundler by default prefers final releases which don't really work. So we need to force them via Gemfile. It doesn't seem like a big deal, it can be removed once there are final releases of those.

Reply:
`io/wait` is already not needed though, could you remove that?

Reply:
Right, `io-wait` was necessary when I first create the PR, but it's no longer a transitive dependency. Let me remove it.

Reply:
Thanks, for `strscan` it currently works by luck in TruffleRuby (see edited comment above), that'll be fixed for future strscan releases with https://github.com/ruby/strscan/pull/35

Reply:
We will not ship JRuby 9.4 with dependencies on the preview gems, so they should go away once we have worked out all issues related to the JRuby versions.